### PR TITLE
nixos/cloudflared: init

### PR DESCRIPTION
**Pull request by piperswe**
###### Motivation for this change

I personally use this module for the services on my home network. It's based on the systemd unit generated by `cloudflared service install`.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by aanderse — CHANGES_REQUESTED** [review]

File: nixos/modules/services/networking/cloudflared.nix [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  cfg = config.services.cloudflared;
+  settingsFormat = pkgs.formats.yaml { };
+in
+{
+  meta.maintainers = with maintainers; [ pmc ];
+
+  options = {
+    services.cloudflared = {
+      enable = mkEnableOption "cloudflared";
+      package = mkOption {
```
[the remark this answers is not in the document]
Comment (lorenzleutgeb) [reply nixos/modules/services/networking/cloudflared.nix:15]: There are many other `*.package` options with the same semantics: https://search.nixos.org/options?query=package

Is there any normative source for preferring overlays over a `.package` option?

File: nixos/modules/services/networking/cloudflared.nix [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        type = types.package;
+        default = pkgs.cloudflared;
+        description = "The cloudflared package to use";
+        example = literalExpression ''pkgs.cloudflared'';
+      };
+      config = mkOption {
```
[the remark this answers is not in the document]
Comment (lorenzleutgeb) [reply nixos/modules/services/networking/cloudflared.nix:21]: I think `.config` is also commonly used and okay.

[the remark this answers is not in the document]
Comment (aanderse) [reply nixos/modules/services/networking/cloudflared.nix:21]: Not commonly, and moreso before https://github.com/NixOS/rfcs/blob/master/rfcs/0042-config-option.md was approved. I prefer if people used the approved convention when one exists.

**Comment by 0x4A6F** [comment]
Closed by #171875.

**Pull request closed without merging**